### ISSUE TITLE
Wells Gardner D9400 Gets Configured Incorrectly

### ISSUE THREAD

**Issue by oscilated** [substring/os#7]
After installation to a hard drive, in setup if a WG-D9400 monitor is chosen, the switchres configuration file gets configured for a D9200.  I don't know if this is just a visual issue or if it matters.  However, what I do know is that X does not get configured to generate a modeline my D9400 can display.

In my particular setup, the D9400 is being driven by an nVidia 7xxx series video card over VGA.  Video setup tool discovers this connection as DVI-I-1 (again, not sure if it matters).  And the video setup tool otherwise fails to detect the monitor (maybe the monitor doesn't support EDID?).

**Comment by substring** [comment]
d9200 instead of d9400 typo will be fixed in the next gatools update (although it's too late for your install for that particular point, you can still manually edit files)

Can you ssh to your box and give me the output of `xrandr -d :0` ? I guess I know what's happening.

By the way, using a Nvidia card is a bad idea, too much tearing. The monitor wasn't detected most probably because its impedance is too low so that your GPU detects it. EDID can help, but the detection tool is designed to force enable analog connectors even if no monitor is found (which must have happened to you and the OS was then configured accordingly)

**Comment by oscilated** [comment]
BTW, I'm terribly sorry for not having mentioned "thank you for all your efforts".  Seriously, I appreciate seeing so many people come together to create such amazing products.  Your time is valuable.  I recognize you chose to donate your time to create a piece of software that primarily benefits others.

Per your request, here's the output with X11 running (I'm going to use a quote block because the code block is stripping line feeds):

> Screen 0: minimum 320 x 200, current 1024 x 768, maximum 4096 x 4096
DVI-I-1 connected primary 1024x768+0+0 (normal left inverted right x axis y axis) 0mm x 0mm
   1024x768      60.00*
   800x600        60.32       56.25
   848x480        60.00
   640x480        59.94
VGA-1 disconnected (normal left inverted right x axis y axis)
TV-1 disconnected (normal left inverted right x axis y axis)

Regarding the nVidia card, I'm not so sure tearing would be the problem.  Admittedly most nVidia cards don't support pixel clocks lower than 5 or so, but super resolutions should solve that issue.

FYI, I'm coming from a 100% custom "hand made from scratch" linux system I compiled 10+ years ago (see [LFS](www.linuxfromscratch.org)).  I was able to get many arcade/console outputs perfect with hand-messaged framebuffer modelines.  But since MAME now requires X11, and OpenGL I can't just compile a new MAME to stay current.  Regardless, coming from my current experience, it's my belief that the video card won't be an issue here.  But then again, I'm willing to be wrong :)

Thanks for the help.  I'm not a programmer, but I can get useful results with Bash scripts, and am very comfortable with Linux.  Hopefully the two of us can get the D9400 working perfectly within GroovyArcade.

**Comment by substring** [comment]
Hopefully I detect GPU manufacturers and set a min dotclock of 25MHz when required. But Nvidia drivers handle vsync pretty bad and have some ugly way to circumvent that (and we need xorg .conf specific parameters in that case. It will force double buffering, so add 1 frame of lag).

In the case of 31kHz monitors without edid, linux doesn't set 640x480, but adds it to a list of default resolutions (1024x768, 800x600, 848x480). To fix that, just add `drm.edid_firmware=DVI-I-1:edid/d9400.bin` to the kernel command line at /boot/syslinux/syslinux.cfg

And, you're welcome, it's a pleasure and great fun to mtaintain this distro :)

**Comment by oscilated** [comment]
Just to be clear (and maybe I'm misinterpreting your comment about 31kHz monitors) a Wells Gardner D9400 is a "tri-sync" monitor capable of 15kHz, 25kHz, and 31kHz modes.

Give me a few minutes to try the kernel command line.  I'll report back.

**Comment by oscilated** [comment]
FYI, here is my existing syslinux.cfg:

> default arch
timeout 0
prompt 0
#UI vesamenu.c32
menu title Groovy Arcade Linux
menu background splash.png
label arch
menu label GroovyArcade
linux /vmlinuz-linux-15khz
append root=/dev/disk/by-label/GA rw quiet rd.udev.log-priority=3 splash mitigations=off audit=0 monitor=d9200 video=DVI-I-1:640x480
initrd /initramfs-linux-15khz.img

Notice the `monitor=d9200`  I've just now changed it to `monitor=d9400` and added `drm.edid_firmware=DVI-I-1:edid/d9400.bin`

**Comment by oscilated** [comment]
BAM!

The modifications to syslinux.cfg mentioned above now result in a useful X11 image.

Thanks for the help.  Is there anything I can do to help you get the monitor detection/selection working correctly (at least for a D9400)?

**Comment by substring** [comment]
the `monitor` isn't used yet. Just here "in case"

I know a d9400 is a trisync, but linux set a 1024x768 by default on it. I'll rewrite the 31kHz part of the kernel parameter after autodetection (it's not just your D9400 but just any 31kHz-capable monitor). So far tri-sync users were very seldom, but as I get more and more support for that, I'll fix it.

**Comment by oscilated** [comment]
Gotcha.

Thanks for the help.  Let me know if you need additional testing.  I'll continue to report issues as I encounter them :)

**Comment by oscilated** [comment]
Just noticed `mame.ini` also got set with `monitor			  d9200`

**Comment by oscilated** [comment]
I have no idea when it happened, but I just today noticed `ga.conf` contained the lines:
```
monitor=d9200
kernel_video_cmdline=monitor=d9200 video=DVI-I-1:640x480
```
and `switchres.conf` contained the lines:
```
monitor=d9200
aspect=auto
```
I could have sworn I manually changed the instances of `monitor=d9200` to `monitor=d9400` a few days ago when we first started looking at this.

Oh well, I'll keep checking those files in the coming days to see if they "automatically" revert back.

**Comment by substring** [comment]
solved in gatools 0.2.5, released with os 2021.03
